src/coprocessor/codec/mysql/duration.rs

```diff
@@ -368,89 +494,174 @@ impl Duration {
             second = Some(block % 100);
         }
 
-        let (hour, minute, second, fraction) = (
+        let (mut hour, mut minute, mut second, mut micros) = (
             hour.unwrap_or(0) + day.unwrap_or(0) * 24,
             minute.unwrap_or(0),
             second.unwrap_or(0),
-            fraction.unwrap_or(0),
+            micros.unwrap_or(0),
         );
 
-        let secs = u64::from(hour) * SECS_PER_HOUR
-            + u64::from(minute) * SECS_PER_MINUTE
-            + u64::from(second);
-
-        // FIXME: We should not round the fractional part with `fsp`
-        let mask = TEN_POW[NANO_WIDTH as usize - fsp as usize - 1];
-        let fraction = (fraction / mask + 5) / 10 * 10 * mask;
+        let secs = hour | minute | second;
 
-        if secs == 0 && fraction == 0 {
+        if secs == 0 && micros == 0 {
```

Comment:
Simply write `if hour == 0 && minute == 0 && second == 0 && micros == 0` should be enough and clear.